src/hyperlink/_url.py

```diff
@@ -501,9 +501,9 @@ def register_scheme(
     """
     text = text.lower()
     if default_port is not None:
-        try:
+        if isinstance(default_port, int) or (isinstance(default_port, str) and default_port.isdigit() and default_port.isascii()):
```

Comment:
Seems like someone could register a negative default port here? Since registering is not performance sensitive in the slightest, we could wrap it in a `str()` and run the same check for consistency?

Reply:
Good catch!
I just pushed up a commit that does exactly what you suggest.
(It also removes some calls to functions not present in python 2, and updates comments/errors)